artichoke-backend/src/intern.rs

```diff
@@ -76,56 +23,85 @@ impl Artichoke {
             Ok(None)
         }
     }
+
+    pub fn intern_bytes_with_trailing_nul<T>(&mut self, bytes: T) -> Result<u32, Exception>
```

Comment:
this change was required to fix a bug when interning the string `"\0"`.

This bug was surfaced by the Ruby specs. 💎